tests/common/async_work_group_copy.h

```diff
@@ -0,0 +1,513 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Common checks for async_work_group_copy and wait_for calls
+//  for nd_item and group tests
+//
+*******************************************************************************/
+
+#ifndef __SYCLCTS_TESTS_COMMON_ASYNC_WORK_GROUP_COPY_H
+#define __SYCLCTS_TESTS_COMMON_ASYNC_WORK_GROUP_COPY_H
+
+#include "../common/common.h"
+#include "../common/once_per_unit.h"
+#include "../common/type_coverage.h"
+#include "../../util/array.h"
+#include "../../util/variadic.h"
+
+/**
+ * @brief Initializes buffer given by pointer and size with series of 1 and 0
+ * @retval Values used for initialization to use as the reference ones
+ * @tparam bufferSize Size of buffer to initialize
+ * @tparam T Deduced type of values to store
+ * @tparam addressSpace Deduced address space
+ * @param ptr Pointer to the start of buffer to initialize
+ */
+template <size_t bufferSize, typename T,
+          cl::sycl::access::address_space addressSpace>
+sycl_cts::util::array<T, bufferSize> create_async_wg_copy_input(
+    cl::sycl::multi_ptr<T, addressSpace> ptr) {
+
+  sycl_cts::util::array<T, bufferSize> result;
+
+  /** We use bool values because bool can be converted to the integral
+   *  and to the floating point types
+   */
```

Comment:
I'm using bool value because it can be implicitly converted to any other type. To use static_cast we should provide some other function so that scalar T and vector T will go through different code path. Say, for `T=vec<bool, 2>` this implementation does work without explicit `T::element_type` usage.
Should I provide some integer sequence for non-boolean types? Or may be should I use some different binary sequence, `0-1-1` for example?